service/src/test/java/uk/nhs/adaptors/gp2gp/common/task/TaskHandlerTest.java

```diff
@@ -306,15 +320,42 @@ public void When_Handle_WithDataAccessResourceFailureException_Expect_ExceptionT
     @SneakyThrows
     public void When_Handle_WithMhsServerErrorException_Expect_ProcessFailed() {
         setUpContinueMessage();
-        doThrow(new MhsServerErrorException("Test exception")).when(taskExecutor).execute(any());
+        Exception exception = new MhsServerErrorException(TEST_EXCEPTION_MESSAGE);
+        doThrow(exception).when(taskExecutor).execute(any());
 
         var result = taskHandler.handle(message);
         assertThat(result).isFalse();
 
-        verify(processingErrorHandler).handleGeneralProcessingError(any());
+        verify(taskErrorHandler).handleProcessingError(eq(exception), any());
 
     }
 
+    @Test
+    @SneakyThrows
+    public void When_Handle_WithGpConnectInvalidException_Expect_ProcessFailed() {
```

Comment:
I have mixed feelings about being specific about `GpConnectInvalidException` within this unit test.  Feels slightly unnecessary (and potentially misleading), given that there is no behaviour for it.

Reply:
Yeah, that makes sense. There are several similar tests that can be reduced to one